language-plutus-napkin/src/Language/PlutusNapkin/Identifier.hs

```diff
@@ -0,0 +1,27 @@
+module Language.PlutusNapkin.Identifier ( -- * Types
+                                          IdentifierState
+                                        -- * Functions
+                                        , newIdentifier
+                                        , emptyIdentifierState
+                                        ) where
+
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.IntMap          as IM
+import qualified Data.Map             as M
+
+-- | An 'IdentifierState' includes a map indexed by 'Int's as well as a map
+-- indexed by 'ByteString's.
+type IdentifierState = (IM.IntMap BSL.ByteString, M.Map BSL.ByteString Int)
```

Comment:
I think, we should give the `Int`s that are unique numbers representing identifiers their own `newtype` and export that abstractly. After all, we don't want clients to perform arithmetic on them 😉